packages/math-input/src/components/input/math-input.tsx

```diff
@@ -328,11 +336,21 @@ class MathInput extends React.Component<Props, State> {
 
     blur: () => void = () => {
         this.mathField.blur();
-        this.props.onBlur && this.props.onBlur();
-        this.setState({focused: false, handle: {visible: false}});
+
+        this.setState((prevState) => {
+            return {
+                showInputFocusStyle: false,
+                handle: {
+                    visible: false,
+                },
+                focused: prevState.focused,
+            };
+        });
     };
 
-    focus: () => void = () => {
+    focus: (setKeypadActive: (keypadActive: boolean) => void) => void = (
```

Comment:
It looks like we use `(keypadActive: boolean) => void` in a few places and this particular type of nested functions is hard to read. I think it's worth making a defined type for this that can be used throughout this file.

Reply:
Ooo good callout! This is all coming from the KeypadContext Provider, so I've made sure to update all the locations to use either `KeypadContextType['setKeypadActive']` or `KeypadContextType['keypadActive']`